shapeshift/web
# chore: simplify zrx swapper

## Description

Removes the following:

- the concept of multiple zrx swappers and combines all evm zrx swappers into a single instance.
- `SwapperType` enum which was used to manage swappers, now using `SwapperName`.
- the requirement for web3 and chainadapters being passed around via `deps`

For context, this change is motivated by incoming architectural changes and getting this in first will simplify the review and code changes somewhat.

## Pull Request Type


## Issue (if applicable)

NA

## Risk

- Risk of broken swappers due to removal of `SwapperType` enum which was used to manage swappers
- Risk of broken 0x swapper.

## Testing

- Check quotes are fetched correctly
- Check 0x trades are working
- Check other swappers are still working

### Engineering

Note that feature flags have changed so you'll need to regenerate if running locally.

### Operations

This change should have no impact on user-facing behaviour, so if anything is different from prod its likely a problem.

## Screenshots (if applicable)
